src/kprofile.cpp

```diff
@@ -18,40 +23,66 @@ KProfile::KProfile(QObject *parent) :
 #endif
 }
 
-#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
-void KProfile::startDownload(QWebEngineDownloadItem* download)
-#else
-void KProfile::startDownload(QWebEngineDownloadRequest* download)
-#endif
+namespace {
+    void setDownloadFilePath(KProfile::WebEngineDownloadType* download, QString filePath)
+    {
+    #if QT_VERSION < QT_VERSION_CHECK(5, 14, 0)
+        download->setPath(filePath);
+    #else
+        download->setDownloadFileName(filePath);
+    #endif
+    }
+}
+
+void KProfile::openFile(WebEngineDownloadType* download)
+{
+    QString defaultFileName = download->url().fileName();
+    QTemporaryFile tempFile(QDir::tempPath() + "/XXXXXX." + QFileInfo(defaultFileName).suffix());
+    tempFile.setAutoRemove(false);
+    if (tempFile.open()) {
+        QString tempFilePath = tempFile.fileName();
+        tempFile.close();
+        setDownloadFilePath(download, tempFilePath);
+        connect(download, &WebEngineDownloadType::finished, [tempFilePath]() {
```

Comment:
This still shouldn't compile under Qt6 (because the signal name is `isFinished()` there).

Reply:
Seems like I can't typedef a signal, have used #define here, is that ok?